.swiftlint.yml

```diff
@@ -0,0 +1,88 @@
+excluded:
+  - Scripts
+  - Package.swift
+  - DocsSnippets
+  - Dependencies
+  - Sources/StreamChatUI/Generated
+  - Sources/StreamChatUI/StreamSwiftyGif
+  - Sources/StreamChatUI/StreamNuke
+  - Sources/StreamChat/StreamStarscream
+  - vendor
+  - UISDKdocumentation
+  - Tests
+  - Sources/StreamChatUI/Deprecations.swift
+  - Sources/StreamChatUI/**/**/AttachmentViewCatalog.swift
+  - Sources/StreamChat/Controllers/DataController.swift
+
+disabled_rules:
+  - large_tuple
+  - multiple_closures_with_trailing_closure
+  - todo
+  - function_parameter_count
```

Comment:
We actually violate a lot this rule in our project, and I think we shouldn't. We should try and extract parameters to structs, to avoid this, and sometimes it is actually a code smell if you have a function with many arguments. Ideally, we would disallow this rule in the places we can't break it right now, but for future functions, we would be mindful when doing this. But maybe this is spread across the codebase right now?

Reply:
Yeah, so the ones which are widespread I was thinking to ignore and create a new ticket to cover those rules specifically?

Reply:
For now, would be nice to just ignore them, and if we think it is necessary we can create tickets in the future for this case 👍